Feature/web/issue list

maong0927 opened the pull request:
### 변경사항
1. 서버에서 가져온 이슈 목록 렌더링 구현 

### 작업 유형
- [x] 신규 기능 추가
- [ ] 버그 수정
- [ ] 리펙토링
- [ ] 문서 업데이트

### 체크리스트
- [x] Merge 하는 브랜치가 올바른가?
- [x] 코딩컨벤션을 준수하는가?
- [x] PR과 관련없는 변경사항이 없는가?
- [x] 내 코드에 대한 자기 검토가 되었는가?
- [ ] 변경사항이 효과적이거나 동작이 작동한다는 것을 보증하는 테스트를 추가하였는가?
- [ ] 새로운 테스트와 기존의 테스트가 변경사항에 대해 만족하는가?

maong0927 committed:
feat    : :zap:️ Lable link 추가

maong0927 committed:
Merge branch 'dev-web' of https://github.com/boostcamp-2020/IssueTracker-15 into feature/web/issueList

maong0927 committed:
style   : :high_heel: issue filter bar, 버튼 크기 변경

1. new issue 버튼에 맞추어 크기 변경

maong0927 committed:
feat    : :zap:️ issue Component 세부사항 추가

1. Assignee, Label 항목 추가

maong0927 committed:
feat    : :zap:️ 전체 issue Fetch 함수 작성

1. 전체 issue 가져오는 fetch 함수 작성
2. label List 에서 초기 렌더링 시 전체 이슈 가져오게 구현

maong0927 committed:
feat    : :zap:️ 서버에서 가져온 issue 목록 렌더링

component 들 props 로 issue 정보 받을 수 있게 함.

maong0927 committed:
style   : :high_heel: z-index 변경

드롭다운이 뒤에 오는 문제를 해결하기 위해 z-index 변경

maong0927 committed:
refactor: :rainbow: Filter 개별 컴포넌트 리팩토링

maong0927 committed:
feat    : :zap:️ milestone 전체 가져오는 api 구현

  1. get 요청으로 마일스톤 전체 가져옴

namda-on commented on web/client/src/components/issue-form/header/index.tsx:
```diff
@@ -2,18 +2,23 @@ import React, { useState } from "react";
-        <FilterListComp title="sort"></FilterListComp>
+        {Filters.map((filter) => {
+          return <FilterListComp title={filter}></FilterListComp>;
+        })}
```
고차함수로 바꾸어주신거 좋네요!

moaikang replied:
```
 {Filters.map((filter) => {
          return <FilterListComp title={filter}></FilterListComp>;
        })}
```

```
{Filters.map((filter) => (
          <FilterListComp title={filter}/>
        )}
```

이런식으로 코드를 줄일 수 있을 거 같아요!

namda-on commented on web/client/src/components/issue-form/issueList/index.tsx:
```diff
@@ -1,13 +1,26 @@
+    };
+    fetchIssues();
+  }, []);
+  console.log(issues);
```
console.log가 남아있네욥 ㅎㅎ ..

maong0927 replied:
ㅎㅎ 얼른 지우겠슴당!

namda-on commented on web/client/src/components/issue-form/issueList/issueComp/index.tsx:
```diff
@@ -2,24 +2,39 @@ import React, { useState, Fragment } from "react";
+import ProfileImage from "../../../profile-image";
+import { getTimeTillNow } from "../../../../lib/dateParser";
+
+interface IssueListProp {
+  info: any;
+}
```
이제 받아오는 info (issue) 의 형식이 정해져있으니 inteface로 type을 구체적으로 정의해줘도 괜찮을것 같아요! 😁

maong0927 replied:
아 넵넵 !! 나중에 고쳐야지 했는데 지금 고쳐야겠네용

namda-on reviewed (approved):
고생하셨습니다!! 너무 좋아요~

maong0927 committed:
feat    : :zap:️ milsteone 목록 가져오기 구현

1. 처음 렌더링 될 때 milestone 목록 가져오게 구현
2. 안쓰는 console.log 제거

maong0927 committed:
feat    : :zap:️ issue 목록 뷰에 마일스톤 이름 추가

1. 각각의 이슈에 마일스톤 이름 추가

maong0927 committed:
feat    : :zap:️ issue상세화면 Link 구현

1. issue 컴포넌트 누르면 상세화면으로 이동

maong0927 committed:
style   : :high_heel: hover 스타일 추가

1. 개별 issue 컴포넌트 hover 시 뒷배경색 바뀌게 변경
2. 개별 버튼 컴포넌트 hover 시 pointer 모양으로 커서 바뀌게 변경

namda-on commented on web/client/src/components/issue-form/issueList/issueComp/index.tsx:
```diff
@@ -21,7 +20,9 @@ function IssueList({ info }: IssueListProp) {
+          <S.IssueInfoLink to={`/issue/${info.id}`}>
+            <S.IssueTitle>{info.title} </S.IssueTitle>
+          </S.IssueInfoLink>
           {info.labels.map((label: any) => {
```
label type은 @types에 label-form 에 만들어놨습니다. 자주 쓰는 타입이니까요 ㅎㅎ 이걸 불러와서
```
labels.map((label : Label ) => {
 
}
```
이런것으로 하시면 좋을 것 같습니다.

maong0927 replied:
오오 ㅎㅎ 감사합니다!

namda-on commented on web/client/src/components/issue-form/issueList/issueComp/style.tsx:
```diff
@@ -49,3 +56,8 @@ export const ProfileImageWrapper = styled.div<{ size: number }>`
   left: ${(props) => props.size}%;
 `;
+
+export const IssueInfoLink = styled(Link)`
+  text-decoration: none;
+  color: #586069;
+`;
```
오 이렇게 할 수 있군요! 😮

maong0927 replied:
근우님께 배웠씁니당!

moaikang commented on web/client/src/components/issue-form/header/index.tsx:
```diff
@@ -1,19 +1,36 @@
+  useEffect(() => {
+    const fetchMilestones = async () => {
+      const newMilestones = await getMilestones();
+      const newMilestonesTitle = newMilestones.map((milestone: any) => {
+        return milestone.title;
+      });
```
```
   const newMilestonesTitle = newMilestones.map((milestone: any) => {
        return milestone.title;
      });
```
를

```
   const newMilestonesTitle = newMilestones.map((milestone: Milestone) => milestone.title);
```

이런식으로 코드를 고칠 수 있을거 같아요!

maong0927 replied:
아 그러네용!!
